pkg/services/ngalert/notifier/images.go

```diff
@@ -20,21 +20,26 @@ func newImageStore(store store.ImageStore) images.ImageStore {
 	}
 }
 
-func (i imageStore) GetImage(ctx context.Context, token string) (*images.Image, error) {
-	image, err := i.store.GetImage(ctx, token)
-	if err != nil {
-		if errors.Is(err, models.ErrImageNotFound) {
-			err = images.ErrImageNotFound
+func (i imageStore) GetImage(ctx context.Context, uri string) (*images.Image, error) {
+	var image *models.Image
+	var err error
+	// Check whether the uri is a token or a URL to know how to query the DB.
+	token := strings.TrimPrefix(uri, "token://")
+	if len(token) < len(uri) {
+		// If the final string is shorter, that means it was prefixed.
+		if image, err = i.store.GetImage(ctx, token); err != nil {
+			return nil, err
 		}
-	}
-	var result *images.Image
-	if image != nil {
-		result = &images.Image{
-			Token:     image.Token,
-			Path:      image.Path,
-			URL:       image.URL,
-			CreatedAt: image.CreatedAt,
+	} else {
+		if image, err = i.store.GetImageByURL(ctx, uri); err != nil {
+			return nil, err
 		}
 	}
-	return result, err
+
```

Comment:
I think this can be simplified even more
```go
if len(token) = len(uri) {
    image, err = i.store.GetImageByURL(ctx, uri)
} else {
    image, err = i.store.GetImage(ctx, token)
}
if  err != nil {
    return nil, err
}
...

Reply:
I did that at first but opted for having the error handling right next to the function call. I'm fine with either approach.